explorer/interpreter/interpreter.cpp

```diff
@@ -2253,19 +2258,25 @@ auto Interpreter::StepStmt() -> ErrorOr<Success> {
         ExpressionCategory expr_category =
             definition.has_init() ? definition.init().expression_category()
                                   : ExpressionCategory::Value;
+        const auto init_location =
+            (act.location_received() && definition.is_returned())
+                ? act.location_received()
+                : act.location_created();
         if (definition.has_init()) {
-          const auto init_location = act.location_created();
           if (expr_category == ExpressionCategory::Initializing &&
-              init_location && heap_.IsInitialized(*init_location)) {
+              init_location && heap_.is_initialized(*init_location)) {
             const auto address = Address(*init_location);
             CARBON_ASSIGN_OR_RETURN(
                 v, heap_.Read(address, definition.source_loc()));
-            CARBON_CHECK(v == act.results()[0]);
+            // CARBON_CHECK(v == act.results()[0]);
```

Comment:
This should either be uncommented again or removed, I'm not sure what's right (maybe you commented it because it's no longer correct, given the cast)
```suggestion
```